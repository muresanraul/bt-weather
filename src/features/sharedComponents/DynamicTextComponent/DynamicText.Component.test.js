import React from 'react';
import renderer from 'react-test-renderer';

import { default as DynamicTextComponent } from './DynamicText.Component';

describe('DynamicTextComponent:', () => {

  it('Component - renders correctly with no properties', () => {
    const dynamicTextComponent = renderer.create(
      <DynamicTextComponent />
    ).toJSON();
    expect(dynamicTextComponent).toMatchSnapshot();
  });

  it('Component - renders correctly with set text property', () => {
    const props = {text: 'text'}
    const imageLoaderComponent = renderer.create(
      <DynamicTextComponent {...props}/>
    ).toJSON();
    expect(imageLoaderComponent).toMatchSnapshot();
  });

  it('Component - renders correctly with set text and fontSize properties', () => {
    const props = {text: 'text', fontSize: '12px'}
    const imageLoaderComponent = renderer.create(
      <DynamicTextComponent {...props}/>
    ).toJSON();
    expect(imageLoaderComponent).toMatchSnapshot();
  });

  it('Component - renders correctly with set text, fontSize and color properties', () => {
    const props = {text: 'text', fontSize: '12px', color: 'red'}
    const imageLoaderComponent = renderer.create(
      <DynamicTextComponent {...props}/>
    ).toJSON();
    expect(imageLoaderComponent).toMatchSnapshot();
  });

  it('Component - renders correctly with set fontSize and color properties but no text property set, ', () => {
    const props = {text: 'text', fontSize: '12px', color: 'red'}
    const imageLoaderComponent = renderer.create(
      <DynamicTextComponent {...props}/>
    ).toJSON();
    expect(imageLoaderComponent).toMatchSnapshot();
  });
});
