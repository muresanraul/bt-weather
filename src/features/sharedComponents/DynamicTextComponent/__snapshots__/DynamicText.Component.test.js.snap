// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicTextComponent: Component - renders correctly with no properties 1`] = `
<div
  className="dynamic-text"
>
  <div
    className="dynamic-text__placeholder"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />
</div>
`;

exports[`DynamicTextComponent: Component - renders correctly with set fontSize and color properties but no text property set,  1`] = `
<div
  className="dynamic-text"
>
  <div
    className="dynamic-text__text"
    style={
      Object {
        "color": "red",
        "fontSize": "12px",
        "justifyContent": undefined,
      }
    }
  >
    text
  </div>
</div>
`;

exports[`DynamicTextComponent: Component - renders correctly with set text and fontSize properties 1`] = `
<div
  className="dynamic-text"
>
  <div
    className="dynamic-text__text"
    style={
      Object {
        "color": undefined,
        "fontSize": "12px",
        "justifyContent": undefined,
      }
    }
  >
    text
  </div>
</div>
`;

exports[`DynamicTextComponent: Component - renders correctly with set text property 1`] = `
<div
  className="dynamic-text"
>
  <div
    className="dynamic-text__text"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
        "justifyContent": undefined,
      }
    }
  >
    text
  </div>
</div>
`;

exports[`DynamicTextComponent: Component - renders correctly with set text, fontSize and color properties 1`] = `
<div
  className="dynamic-text"
>
  <div
    className="dynamic-text__text"
    style={
      Object {
        "color": "red",
        "fontSize": "12px",
        "justifyContent": undefined,
      }
    }
  >
    text
  </div>
</div>
`;
